Week3/homework/6-step3.js

```diff
@@ -4,3 +4,19 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 // add your solution here, or add a comment on how you would tackle this problem
+
+let printAllTheItems = [];
+
+function printTheItemsOfArray(arr) {
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+
+      printAllTheItems += (arr[i][j] + ' ');
+    }
+  }
+  console.log(printAllTheItems);
+}
+
```

Comment:
While this works for the given examples, will it work for this nested array?
```
const arrNd = [[[[1, 2]], [3, 4], [[[5, 6], [7, 8]], [9, 10]], [11, 12]], [[[[[[13, 14]]]]]]];
```
- I would suggest looking into `recursion`.  It takes some getting used to, but once you get the idea, for the same amount of code it would handle any depth of nesting.
An example that does not mutate any variables:
```
function printAllTheItems(arr){ 
  let values = [];
  for (let i = 0; i < arr.length; i++){
    if (Array.isArray(arr[i])) {
      values = values.concat(printAllTheItems(arr[i]));
    } else {
      values = values.concat([arr[i]]);
    }
  }
  return values;
};
```

- I would also have kept the array where you accumulated the values in the function.  Otherwise it is not safe from being changed by other code.
- The name of the array that accumulates the values is also confusing.  I would be inclined to call it `allTheValues` as it does not `print` anything, just holds values.  The function does the printing.
In general, you want to name data using nouns, and functions as verbs.